src/mon/MonCap.h

```diff
@@ -196,13 +196,22 @@ struct MonCap {
   }
 
   bool fs_name_capable(string_view fs_name, __u8 mask) {
-    for (auto& g: grants) {
+    for (auto& g : grants) {
       if (g.is_allow_all()) {
 	return true;
       }
-      if (g.fs_name.empty() || g.fs_name == fs_name) {
-	if (mask & g.allow) {
+
+      if ((g.fs_name.empty() || g.fs_name == fs_name) && (mask & g.allow)) {
 	  return true;
+      }
+
+      if (not g.profile.empty()) {
+	for (auto& pg : g.profile_grants) {
+	  if ((pg.service == "fs" || pg.service == "mds") &&
```

Comment:
The service has to be taken as function argument and compared against pg.service. 
The problem with this approach is that if the "mgr" profile is something like below
{
service=mds, allow=MON_CAP_W
service=fs, allow=MON_CAP_R
}
and a fs command like "fs fail" should not be allowed since "fs" service has only read caps. This condition check allows that command as part mds service check.

Reply:
> and a fs command like "fs fail" should not be allowed since "fs" service has only read caps. This condition check allows that command as part mds service check.

This permission does but the one below it - `pg.allow & mask` - won't.